explainer.md

```diff
@@ -1,624 +1,397 @@
+
+When a developer offers part of the page for transitions, there are two modes they can choose from:
+
+### As a single texture
+
+The entire painting of the element is captured, including things which appear outside of its bounding box such as shadows and blurs, as a single texture at device-pixel resolution.
+
+https://user-images.githubusercontent.com/93594/141118353-d62d19a1-0964-4fa0-880f-bdde656ce899.mp4
+
+- Open question: Which effects from a parent element are baked into this texture? E.g. things like `filter`, `opacity`, `transform`.
+
+Capturing as a texture avoids the interactivity risks, complexities, and memory impact of fully preserving these parts of Page-A as live DOM.
+
+The root is always captured in this way, and is also clipped to the viewport, as capturing the entire page would take an enormous amount of memory in some cases.
+
+- Open question: Should we have a way to expand this area for particular transitions? For example, transitions that involve vertical movement?
+- Open question: Other elements can also be massive. Do we need a way to limit and control the captured size of those?
+
+This mode works great for the share button and the root, as their transitions can be represented by simple transforms. However, the header changes size without stretching its shadow, and the content of the header moves independently. There's another mode for that:
+
+### As a container + child texture
+
+In this mode the computed styles of the element are copied over, so they can be re-rendered beyond just transforming a texture. This allows the developer to animate properties such as width and height, borders, background color… anything that can be animated with CSS. It also allows the developer to create animations that clip child content, via `overflow: hidden`.
+
+The children of the element (including pseudos and text nodes) are captured into a single texture that can be animated independently.
+
+https://user-images.githubusercontent.com/93594/141118395-8d65da49-a5ab-41c6-8458-917e55d4b77b.mp4
+
+A mode like this is unnecessary complexity for the share button in the example transition, but gives the developer the freedom they need for the header transition.
+
+- Open question: Do we need this for 'v1'?
+
+### Nested parts
+
+In the example transition, the content of the header cross-fades from Page-A to Page-B. An even smoother transition could be achieved by also animating the site title and avatar 'chip' independently. However, to reduce the scope of 'v1', offering parts nested in other offered parts is not allowed.
+
+- Open question: If the developer tries to do this, what should happen? Is the nested offering ignored? Is the whole transition abandoned?
+
+## Part 2: The preparation
+
+The state changes over to Page-B, and Page-A is gone aside from the parts it offered.
```

Comment:
I said navigation to highlight that during the navigation sequence the browser starts to fetch page-B before page-A is hidden. And at some point the browser decides to hide page-A and show page-B instead. This state will need to be cached between fetching page-B and hiding page-A.

I had originally talked about events from an MPA perspective (navigation, load, ready for first render) and then mentioned that we'll need APIs for developers to provide the equivalent events explicitly for SPA. But can call this "when a transition is initiated" if you wanted to avoid MPA specific terms here.